extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -127,23 +76,35 @@ func CreateRKE1Snapshot(client *rancher.Client, clusterName string) error {
 			if snapshotObj.State != active {
 				return false, nil
 			}
+
+			snapshotTime, err := time.Parse(time.RFC3339, snapshot.Created)
+			if err != nil {
+				return false, err
+			}
+
+			// time.Parse doesn't include nanoseconds, but time.Now() does. Rounding up by 1 Second.
+			snapshotTime = snapshotTime.Add(time.Duration(oneSecond))
+
+			if snapshotTime.Compare(updateTimestamp) > -1 {
```

Comment:
Have you tested with multiple snapshots for both local and S3 snapshots? Just to validate this is consistently working in that scenario.

Reply:
yes, tested with local and s3, I'll provide some jenkins jobs in the next day or so.